samples/packages/lake-formation-tag-sync/src/main/kotlin/com/atlan/pkg/lftag/LakeTagSynchronizer.kt

```diff
@@ -0,0 +1,107 @@
+        val assetPrefix = Utils.getOrDefault(config.assetsPrefix, "")
+        val batchSize = Utils.getOrDefault(config.batchSize, 20)
+        var results: ImportResults? = null
+
+        val mapper = jacksonObjectMapper()
+
+        if (importType != "CLOUD") {
+            logger.error { "Direct file upload(s) are not supported at this time." }
+            return false
+        }
+        val files = Utils.getInputFiles(
+            "",
+            outputDirectory,
+            false,
+            assetPrefix,
+        )
+        val tagFileNames = mutableListOf<String>()
+        val connectionMap = mutableMapOf<String, String>()
+        val metadataMap = mutableMapOf<String, String>()
+        files.forEach { fileName ->
+            when (fileName) {
+                "$outputDirectory/connection_map.json" -> {
+                    val jsonString: String = File(fileName).readText(Charsets.UTF_8)
+                    connectionMap += mapper.readValue<Map<String, String>>(jsonString)
+                }
+
+                "$outputDirectory/metadata_map.json" -> {
+                    val jsonString: String = File(fileName).readText(Charsets.UTF_8)
+                    metadataMap += mapper.readValue<Map<String, String>>(jsonString)
+                }
+                else -> {
+                    if (File(fileName).nameWithoutExtension.startsWith(TAG_FILE_NAME_PREFIX)) {
+                        tagFileNames.add(fileName)
+                    } else {
+                        logger.warn { "Skipping $fileName because it doesn't start with $TAG_FILE_NAME_PREFIX." }
+                    }
+                }
+            }
+        }
+        if (connectionMap.isEmpty()) {
+            logger.error { "The file connection_map.json must be provided." }
+            return false
+        }
+        if (metadataMap.isEmpty()) {
+            logger.error { "The file metadata_map.json must be provided." }
+            return false
+        }
+        if (tagFileNames.isEmpty()) {
+            logger.error { "You must provide at least one json file containing Lake Tag data." }
+            return false
+        }
+        val csvProducer = CSVProducer(connectionMap, metadataMap, outputDirectory)
+        tagFileNames.forEach { tagFileName ->
+            val csvFileName = tagFileName.replace("$outputDirectory/", "").replace(".json", ".csv")
+            csvProducer.transform(tagFileName, csvFileName)
+            val importConfig = AssetImportCfg(
+                assetsFile = "$outputDirectory/$csvFileName",
+                assetsUpsertSemantic = "update",
+                assetsFailOnErrors = failOnErrors,
+                assetsBatchSize = batchSize,
+                assetsFieldSeparator = ",",
+            )
+            val result = Importer.import(importConfig, outputDirectory)?.combinedWith(results)
+            if (result != null) {
+                results = result.combinedWith(result)
+            }
```

Comment:
This is pretty confusing to follow... This is 1 more line of code, but I think easier to understand (helps to change the variable name, too, as very hard to see difference between `result` and `results` and super easy to make typos)?

```
val result = Importer.import(importConfig, outputDirectory)
if (combinedResults == null) {
    combinedResults = result
} else if (result != null) {
    combinedResults = combinedResults.combinedWith(result)
}
```

Reply:
I know @cmgrote. I tried something similar before. When I try your recommendation I get the following error:
`Smart cast to 'ImportResults' is impossible, because 'combinedResults' is a local variable that is captured by a changing closure:104`
Where 104 refers to `combinedResults = combinedResults.combinedWith(result)`